### ISSUE TITLE
vars-on-top conflicts with 'use strict'

### ISSUE THREAD

**Issue by lo1tuma** [eslint/eslint#1235]
There is a failing test-case for `vars-on-top` which I think should be valid:

``` js
"use strict";
var first = 0;
function foo() {
    first = 2;
}
```
